src/assets/scripts/client/airline/AirlineModel.js

```diff
@@ -207,40 +196,20 @@ export default class AirlineModel extends BaseModel {
 
     // TODO: the logic here can be simplified.
     /**
-     * Create a flight number/identifier
+     * Creates a flight number/identifier
      *
      * This method should only be called from the `AircraftController` so the controller
      * can guarantee unique `flightNumbers` across all `AirlineModels`.
      *
      * @for AirlineModel
      * @method generateFlightNumber
      * @return flightNumber {string}
-     */
+    */
     generateFlightNumber() {
-        let flightNumber = '';
-        let list = '0123456789';
-
-        // Start with a number other than zero
-        flightNumber += choose(list.substr(1));
-
-        if (!this.flightNumberGeneration.alphaNumeric) {
-            for (let i = 1; i < this.flightNumberGeneration.length; i++) {
-                flightNumber += choose(list);
-            }
-
-            return flightNumber;
-        }
-
-        // TODO: why `this.flightNumberGeneration.length - 3`?  enumerate the magic number.
-        for (let i = 0; i < this.flightNumberGeneration.length - 3; i++) {
-            flightNumber += choose(list);
-        }
-
-        list = 'abcdefghijklmnopqrstuvwxyz';
+        const flightNumber = flightNumberBuilder(this.flightNumberGeneration.callsignFormats);
```

Comment:
@n8rzz your opinion?

Reply:
@erikquinn its in a file at my request.  

My reasoning there is that the generation of flight numbers _could_ contain a fair amount of logic/documentation.  Additionally, the `AirlineModel` as a class shouldn't really care _how_ a flight number is generated, only that one is needed based on rules supplied.

This gives us a way to break out that logic to keep the `AirlineModel` smaller with tighter concerns.

Reply:
> This gives us a way to break out that logic to keep the AirlineModel smaller with tighter concerns.

> AirlineModel as a class shouldn't really care how a flight number is generated, only that one is needed based on rules supplied.

I would contend that the sole purpose of the `AirlineModel` would be to generate flight numbers for aircraft spawning. And since this class is definitely the only user of that logic, it seems strange, especially for this particular case that we would want to take the sole task of a simple class and move its logic elsewhere.

If this was at your request, I'll definitely go along with it, but I am trying to get a sense of why we would do this here when we haven't in _much_ larger and more complex classes.